src/jit/codegenlinear.h

```diff
@@ -183,15 +183,18 @@
     void                genJmpMethod(GenTreePtr jmp);
 
 #if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
-    void genGetStructTypeSizeOffset(const SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR& structDesc,
-                                    var_types* type0,
-                                    var_types* type1,
-                                    emitAttr* size0,
-                                    emitAttr* size1,
-                                    unsigned __int8* offset0,
-                                    unsigned __int8* offset1);
+    void getGetStructTypeOffset(const SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR& structDesc,
+                                var_types* type0,
+                                var_types* type1,
+                                unsigned __int8* offset0,
+                                unsigned __int8* offset1);
 
     bool                genStoreRegisterReturnInLclVar(GenTreePtr treeNode);
+    void                getReturnRegistersBasedOnTypes(const SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR& structDesc, 
```

Comment:
How about naming this as getStructReturnRegs()?

This routine will return RAX <- if struct size <=8 bytes or >16 bytes
RAX, RDX if struct size > 8 and <= 16